src/argilla/training/transformers.py

```diff
@@ -322,8 +390,24 @@ def compute_metrics(p):
                 }
 
             func = compute_metrics
+        elif AutoModelForQuestionAnswering:
+            f1 = evaluate.load("f1")
+
+            def compute_metrics_question_answering(pred):
+                squad_labels = pred.label_ids
+                squad_preds = pred.predictions.argmax(-1)
+
+                # Calculate Exact Match (EM)
+                em = sum([1 if p == l else 0 for p, l in zip(squad_preds, squad_labels)]) / len(squad_labels)
+
+                # Calculate F1-score
+                f1_score = f1(squad_labels, squad_preds, average="macro")
+
+                return {"exact_match": em, "f1": f1_score}
```

Comment:
```suggestion
                labels = pred.label_ids
                preds = pred.predictions.argmax(-1)

                # Calculate Exact Match (EM)
                em = sum([int(p == l) for p, l in zip(preds, labels)]) / len(labels)

                # Calculate F1-score
                f1_score = f1(labels, preds, average="macro")

                return {"exact_match": em, "f1": f1_score}
```
Removed "squad" from the naming, and `1 if p == l else 0` is now just done with `int()`.

Reply:
Is the F1 computed for QnA really macro?